protobuf/v1/mayastor.proto

```diff
@@ -7,68 +7,37 @@ syntax = "proto3";
 
 package mayastor.v1;
 
-// Means no arguments or no return value.
-message Null {}
-
-// Anything what follows here are private interfaces used for interacting with
-// mayastor outside the scope of CSI.
+import "google/protobuf/struct.proto";
+import "bdev_json.proto";
 
+//
+// Service to operate directly on block devices
 service BdevRpc {
-  rpc List(Null) returns (Bdevs) {}
-  rpc Create(BdevUri) returns (CreateReply) {}
-  rpc Destroy(BdevUri) returns (Null) {}
-  rpc Share(BdevShareRequest) returns (BdevShareReply) {}
-  rpc Unshare(CreateReply) returns (Null) {}
-}
-
-message BdevShareRequest {
-  string name = 1;
-  string proto = 2;
-}
-
-message BdevShareReply {
-  string uri = 1;
-}
-
-message Bdev {
-  string name = 1;
-  string uuid = 2;
-  uint64 num_blocks = 3;
-  uint32 blk_size = 5;
-  bool claimed = 6;
-  string claimed_by = 7;
-  string aliases = 8;
-  string uri = 9;
-  string product_name = 10;
-  string share_uri = 11;
-
-}
-
-message Bdevs {
-  repeated Bdev bdevs = 1;
-}
-
-message BdevUri {
-  string uri = 1;
-}
-
-message CreateReply {
-  string name = 1;
+  // list the block devices
+  rpc List(NullableString) returns (ListBdevResponse) {}
+  // create a new block device using a uri scheme
+  rpc Create(BdevRequest) returns (BdevResponse) {}
+  // destroys the block device
+  rpc Destroy(BdevRequest) returns (Null) {}
+  // share the block device
+  rpc Share(ShareRequest) returns (ShareResponse) {}
+  // unshare the block device
+  rpc Unshare(UnshareRequest) returns (Null) {}
 }
 
 
-// SPDK json-rpc proxy service
-
 service JsonRpc {
   // Call a (SPDK) json-rpc method
-  rpc JsonRpcCall (JsonRpcRequest) returns (JsonRpcReply) {}
+  rpc JsonRpcCall (JsonRpcRequest) returns (JsonRpcResponse) {}
 }
 
-message JsonRpcRequest {
-  string method = 1;
-  string params = 2;
+message Null {
+  google.protobuf.NullValue null = 1;
```

Comment:
@gila This is the Null message you wanted on `protobuf.NullValue` right?

Reply:
well look here:

``` rpc Unshare(UnshareRequest) returns (Null) {}```

Unshare returns "nothing" it says... So can we simply do returns() ? Some goes for Destroy etc.

Reply:
>  So can we simply do returns() 

This is not the right way in grpc. The proto file wont compile. It expects a return message. 
```
rpc Unshare(UnshareRequest) returns () {}
```

we could use the `google.protobuf.Empty` and write like 
```
rpc Unshare(UnshareRequest) returns (google.protobuf.Empty) {}
```